From e59e3d5ed5325aa1672c08ef643cad6f6d71b2c2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marco=20Trevisan=20=28Trevi=C3=B1o=29?= <mail@3v1n0.net>
Date: Wed, 14 Aug 2019 13:16:31 +0200
Subject: [PATCH 1/5] general: Pass the actor to PopupManuManager constructor
 in new shell

GNOME Shell 3.34 needs an actor as PopupMenuManager argument, so fix this by
passing the actual object in such case.
---
 desktopGrid.js | 4 +++-
 fileItem.js    | 5 ++++-
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/desktopGrid.js b/desktopGrid.js
index a2d1f12..686a6ca 100644
--- a/desktopGrid.js
+++ b/desktopGrid.js
@@ -507,8 +507,10 @@ var DesktopGrid = class {
     }
 
     _addDesktopBackgroundMenu() {
+        let popupManagerArgs = ExtensionUtils.versionCheck(['3.28', '3.30', '3.32'], Config.PACKAGE_VERSION) ?
+            { actor: this.actor } : this.actor;
         this.actor._desktopBackgroundMenu = this._createDesktopBackgroundMenu();
-        this.actor._desktopBackgroundManager = new PopupMenu.PopupMenuManager({ actor: this.actor });
+        this.actor._desktopBackgroundManager = new PopupMenu.PopupMenuManager(popupManagerArgs);
         this.actor._desktopBackgroundManager.addMenu(this.actor._desktopBackgroundMenu);
 
         this.actor.connect('destroy', () => {
diff --git a/fileItem.js b/fileItem.js
index 0c6a54d..164886b 100644
--- a/fileItem.js
+++ b/fileItem.js
@@ -30,6 +30,7 @@ const GnomeDesktop = imports.gi.GnomeDesktop;
 const Mainloop = imports.mainloop;
 const Signals = imports.signals;
 
+const Config = imports.misc.config;
 const Background = imports.ui.background;
 const Main = imports.ui.main;
 const PopupMenu = imports.ui.popupMenu;
@@ -585,8 +586,10 @@ var FileItem = class {
 
     _recreateMenu() {
         this._removeMenu();
+        let popupManagerArgs = ExtensionUtils.versionCheck(['3.28', '3.30', '3.32'], Config.PACKAGE_VERSION) ?
+            { actor: this.actor } : this.actor;
 
-        this._menuManager = new PopupMenu.PopupMenuManager({ actor: this.actor });
+        this._menuManager = new PopupMenu.PopupMenuManager(popupManagerArgs);
         let side = St.Side.LEFT;
         if (Clutter.get_default_text_direction() == Clutter.TextDirection.RTL)
             side = St.Side.RIGHT;
-- 
2.22.0


From 83f4b8f672e31c204486473f51ee1b41e5980156 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marco=20Trevisan=20=28Trevi=C3=B1o=29?= <mail@3v1n0.net>
Date: Wed, 14 Aug 2019 13:20:01 +0200
Subject: [PATCH 2/5] createFolderDialog: Use native GObject for new Shell 3.34
 versions

ModalDialog.ModalDialog is a native object in newer shell versions, so we need
to inherit from it as native object.

Do this only for newer shell versions, while keep the native JS Object otherwise.
---
 createFolderDialog.js | 27 ++++++++++++++++++++++-----
 1 file changed, 22 insertions(+), 5 deletions(-)

diff --git a/createFolderDialog.js b/createFolderDialog.js
index f3e40e9..26aecb3 100644
--- a/createFolderDialog.js
+++ b/createFolderDialog.js
@@ -30,16 +30,17 @@ const ExtensionUtils = imports.misc.extensionUtils;
 const Me = ExtensionUtils.getCurrentExtension();
 const DesktopIconsUtil = Me.imports.desktopIconsUtil;
 const Extension = Me.imports.extension;
+const Config = imports.misc.config;
 
 const _ = Gettext.gettext;
 
 const DIALOG_GROW_TIME = 0.1;
 
-var CreateFolderDialog = class extends ModalDialog.ModalDialog {
-
-    constructor() {
-        super({ styleClass: 'create-folder-dialog' });
+const DIALOG_INIT_PROPS = { styleClass: 'create-folder-dialog' };
 
+let CreateFolderDialogClass = class CreateFolderDialog extends ModalDialog.ModalDialog {
+    _init() {
+        super._init(DIALOG_INIT_PROPS);
         this._buildLayout();
     }
 
@@ -161,4 +162,20 @@ var CreateFolderDialog = class extends ModalDialog.ModalDialog {
         this._createButton.reactive = is_valid;
     }
 };
-Signals.addSignalMethods(CreateFolderDialog.prototype);
+
+var CreateFolderDialog;
+
+if (ExtensionUtils.versionCheck(['3.28', '3.30', '3.32'], Config.PACKAGE_VERSION)) {
+    CreateFolderDialog = class extends CreateFolderDialogClass {
+        constructor() {
+            super(DIALOG_INIT_PROPS);
+            this._buildLayout();
+        }
+    };
+    Signals.addSignalMethods(CreateFolderDialogClass.prototype);
+} else {
+    CreateFolderDialog = GObject.registerClass({
+        GTypeName: 'DesktopIcons_CreateFolderDialog',
+        Signals: { 'response': { param_types: [GObject.TYPE_STRING] } }
+    }, CreateFolderDialogClass);
+}
-- 
2.22.0


From f43f649a984900531dbf416b9aec560a66a03c1d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marco=20Trevisan=20=28Trevi=C3=B1o=29?= <mail@3v1n0.net>
Date: Wed, 14 Aug 2019 13:21:09 +0200
Subject: [PATCH 3/5] desktopGrid: Don't emit deprecation warnings when using
 native objects

Starting from shell 3.34, BoxPointer and MenuItems are native objects, so we
should not sue their actor property not to cause warnings.
---
 desktopGrid.js | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/desktopGrid.js b/desktopGrid.js
index 686a6ca..75893f1 100644
--- a/desktopGrid.js
+++ b/desktopGrid.js
@@ -308,8 +308,12 @@ var DesktopGrid = class {
     }
 
     _syncUndoRedo() {
-        this._undoMenuItem.actor.visible = DBusUtils.NautilusFileOperationsProxy.UndoStatus == UndoStatus.UNDO;
-        this._redoMenuItem.actor.visible = DBusUtils.NautilusFileOperationsProxy.UndoStatus == UndoStatus.REDO;
+        let useItemActors = !ExtensionUtils.versionCheck(['3.28', '3.30', '3.32'], Config.PACKAGE_VERSION);
+        let undoActor = useItemActors ? this._undoMenuItem : this._undoMenuItem.actor;
+        let redoActor = useItemActors ? this._redoMenuItem : this._redoMenuItem.actor;
+
+        undoActor.visible = DBusUtils.NautilusFileOperationsProxy.UndoStatus == UndoStatus.UNDO;
+        redoActor.visible = DBusUtils.NautilusFileOperationsProxy.UndoStatus == UndoStatus.REDO;
     }
 
     _undoStatusChanged(proxy, properties, test) {
@@ -614,7 +618,11 @@ var RenamePopup = class {
         renameContent.add_child(renameButtonsBox);
 
         this._boxPointer = new BoxPointer.BoxPointer(St.Side.TOP, { can_focus: false, x_expand: false });
-        this.actor = this._boxPointer.actor;
+        if (ExtensionUtils.versionCheck(['3.28', '3.30', '3.32'], Config.PACKAGE_VERSION))
+            this.actor = this._boxPointer.actor;
+        else
+            this.actor = this._boxPointer;
+
         this.actor.style_class = 'popup-menu-boxpointer';
         this.actor.add_style_class_name('popup-menu');
         this.actor.visible = false;
-- 
2.22.0


From 4c2ff6a4a29a05822ce1c90003c34c59ea7b60ce Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marco=20Trevisan=20=28Trevi=C3=B1o=29?= <mail@3v1n0.net>
Date: Wed, 14 Aug 2019 13:21:38 +0200
Subject: [PATCH 4/5] metadata: Prettify JSON code

Makes the metadata more readable and easier to see in diffs
---
 metadata.json | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/metadata.json b/metadata.json
index b8e8187..072917f 100644
--- a/metadata.json
+++ b/metadata.json
@@ -1 +1,8 @@
-{"name": "Desktop Icons", "description": "Add icons to the desktop", "uuid": "desktop-icons@csoriano", "shell-version": ["3.30.0"]}
+{
+    "name": "Desktop Icons",
+    "description": "Add icons to the desktop",
+    "uuid": "desktop-icons@csoriano",
+    "shell-version": [
+        "3.30.0"
+    ]
+}
-- 
2.22.0


From 5360d6fcbbe4793f3d356e4dba41cf613aa28999 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marco=20Trevisan=20=28Trevi=C3=B1o=29?= <mail@3v1n0.net>
Date: Wed, 14 Aug 2019 13:23:15 +0200
Subject: [PATCH 5/5] metadata: Support gnome-shell 3.32 and 3.34

---
 metadata.json | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/metadata.json b/metadata.json
index 072917f..361943f 100644
--- a/metadata.json
+++ b/metadata.json
@@ -3,6 +3,8 @@
     "description": "Add icons to the desktop",
     "uuid": "desktop-icons@csoriano",
     "shell-version": [
-        "3.30.0"
+        "3.30.0",
+        "3.32.0",
+        "3.34.0"
     ]
 }
-- 
2.22.0

